packit_service/worker/handlers/open_scan_hub.py

```diff
@@ -0,0 +1,125 @@
+from packit_service.worker.events import (
+    OpenScanHubTaskFinishEvent,
+)
+from packit_service.worker.handlers.abstract import (
+    RetriableJobHandler,
+    TaskName,
+    configured_as,
+    reacts_to,
+)
+from packit_service.worker.helpers.build import CoprBuildJobHelper
+from packit_service.worker.helpers.scan import ScanHelper
+
+from packit_service.worker.handlers.mixin import (
+    ConfigFromEventMixin,
+)
+from packit_service.worker.result import TaskResults
+from packit_service.worker.mixin import (
+    LocalProjectMixin,
+    PackitAPIWithUpstreamMixin,
+)
+
+from packit_service.worker.reporting import BaseCommitStatus
+
+logger = logging.getLogger(__name__)
+
+
+@configured_as(job_type=JobType.copr_build)
+@reacts_to(OpenScanHubTaskFinishEvent)
+class OpenScanHubTaskFinishHandler(
+    RetriableJobHandler,
+    LocalProjectMixin,
+    ConfigFromEventMixin,
+    PackitAPIWithUpstreamMixin,
+):
+    task_name = TaskName.openscanhub_task_finish
+
+    @staticmethod
+    def get_checkers() -> Tuple[Type[Checker], ...]:
+        return ()
+
+    def run(self) -> TaskResults:
+        task_id = self.data.event_dict["task_id"]
+        event = self.data.to_event()
+        if not event.scan or not event.build:
+            return TaskResults(
+                success=False,
+                details={
+                    "msg": f"Scan {task_id} not found or not associated with a Copr build"
+                },
+            )
+        elif not self.job_config:
+            return TaskResults(
+                success=False,
+                details={
+                    "msg": (
+                        "No job configuration found for "
+                        f"openscanhub_task_finish in {self.project.repo}"
+                    )
+                },
+            )
+
+        branches = aliases.get_build_targets(
+            *self.job_config.targets,
+        )
+        if "fedora-rawhide-x86_64" not in branches:
+            return TaskResults(
+                success=False,
+                details={
+                    "msg": "Skipping job configuration with no fedora-rawhide-x86_64 target."
+                },
+            )
+
+        build_helper = CoprBuildJobHelper(
+            service_config=self.service_config,
+            package_config=self.package_config,
+            project=self.project,
+            metadata=self.data,
+            db_project_event=self.data.db_project_event,
+            job_config=self.job_config,
+            celery_task=self.celery_task,
+        )
+
+        scan_helper = ScanHelper(copr_build_helper=build_helper, build=event.build)
+
+        external_links = {
+            "Added issues": event.issues_added_url,
+            "Fixed issues": event.issues_fixed_url,
+            "Scan results": event.scan_results_url,
+        }
+
+        # TODO: probably we need a babysit task for when the build is not finished yet
+        if event.build.status == "success":
```

Comment:
shouldn't we be checking the status of the scan and not build? We submit the scans just for successful build, so there should be no need for this

Reply:
good point, @siteshwar could you please provide more details how the message looks when the build is failed/cancelled/interrupted vs when it finishes successfully? Is it detectable by the result files (like added, fixed) not being present?

Reply:
> good point, @siteshwar could you please provide more details how the message looks when the build is failed/cancelled/interrupted vs when it finishes successfully? Is it detectable by the result files (like added, fixed) not being present?

These are the topics for tasks that do not finish successfully:
```
org.fedoraproject.stg.openscanhub.task.cancel
org.fedoraproject.stg.openscanhub.task.fail
org.fedoraproject.stg.openscanhub.task.interrupt
```

And the body contains the build id.

Other than that, you may also want to process:
```
org.fedoraproject.stg.openscanhub.task.open
```

which means a worker has started running the task.

Reply:
I have summarized the steps again in https://github.com/packit/packit-service/issues/2543#issuecomment-2417233326

Reply:
Thanks a lot @siteshwar now it is far more clear to me. I think it will not be a big deal to handle the other messages as well. Still I want to ask you if there is a reason why you created different messages? I would have expected a `started` and a `finished` task message but not a `cancel or fail or interrupt` message. I see the *cancel or fail or interrupt* more as states for the *finished* action. For example for the [copr.build.end](https://apps.fedoraproject.org/datagrepper/v2/id?id=353d4c59-eafa-4043-aa7e-6386e7bb595b&is_raw=true&size=extra-large) message there is a `status` field and there are just the `copr.build.start` and `copr.build.end` topics/messages. As I wrote, we can deal with more topics/messages but maybe you also prefer having just one finish message?

Reply:
One more thing, if we will have all this messages I suppose I will never receive a `finish` message together with an `interrupt or cancel or fail` message for the same `task_id`, right? Otherwise, if the order of the messages is not preserved in our workers, I could override a failed scan as successful.

Reply:
> As I wrote, we can deal with more topics/messages but maybe you also prefer having just one finish message?

I am just reflecting the different states of tasks in the messages as Kobo.

@kdudka Shall we change the message topics as Copr or keep them as it is?

Reply:
Yep, the proposed change sounds reasonable.  Let's implement the messaging such that it is better aligned with other services that Packit listens to.